design_drafts/ros2_control_documentation.rst

```diff
@@ -0,0 +1,273 @@
+
+The execution of the control-loop is managed by the CM's ``update()`` method.
+The method reads data from the hardware components, updates outputs of all active controllers, and writes the result to the components.
+
+Resource Manager
+----------------
+The `Resource Manager`_ (RM) abstracts physical hardware and its drivers (called *hardware components*) for the ros2_control framework.
+The RM loads the components using ``pluginlib``-library, manages their lifecycle and components' state and command interfaces.
+This abstraction provided by RM enables re-usability of implemented hardware components, e.g., robot and gripper, without any implementation and flexible hardware application for state and command interfaces, e.g., separate hardware/communication libraries for motor control and encoder reading.
+
+In the control loop execution, the RM's ``read()`` and ``write()`` methods deal with communication to the hardware components.
+
+.. _overview-controllers:
+Controllers
+-----------
+The controllers in the ros2_control framework have the same functionality as defined in the control theory. They compare the reference value with the measured output and, based on this error, calculate a system's input (for more details, visit `Wikipedia <https://en.wikipedia.org/wiki/Control_theory>`_).
+The controlles are objects derived from `ControllerInterface`_ (``controller_interface`` package in `ros-controls/ros2_control`_) and exported as plugins using ``pluginlib``-library.
+For example of on controller check `ForwardCommandController implementation`_ in the `ros-controls/ros2_controllers`_ repository.
+The controllers' lifecycle is based on the `LifecycleNode-Class`_ implementing the state machine as described in the `Node Lifecycle Design`_ document.
+
+When executing the control-loop ``update()`` method is called.
+The method can access the latest hardware states and enables the controller to write the hardware's command interfaces.
+
+User Interfaces
+---------------
+Users interact with the ros2_control framework using `Controller Manager`_'s services.
+Those can be used directly or through the "cli"-interface (base command: ``ros2 control``).
+The "cli"-interface is more user-friendly for direct interaction, i.e., outside of a node.
+
+For a list of services and their definitions, check the ``srv`` folder in the `controller_manager_msgs`_ package.
+
+For the description of the "cli"-interface, see the ``README.md`` file of the `ros2controlcli`_ package.
+
+
+Hardware Components
+===================
+The *hardware components* realize communication to physical hardware and represent its abstraction in the ros2_control framework.
+The components have to be exported as plugins using ``pluginlib``-library.
+The `Resource Manager`_ dynamically loads those plugins and manages their lifecycle.
+
+There are three basic types of components:
+
+Actuators
```

Comment:
It would be more intuitive for a first reader to have the order flipped here: system, sensors, actuators --- as this is the order of importance for the majority of users

Reply:
You are right, I was just listed it alphabetically...